src/main/java/com/example/music/backend/verification/controller/RegistrationController.java

```diff
@@ -0,0 +1,57 @@
+package com.example.music.backend.verification.controller;
+
+import com.example.music.backend.user.domain.User;
+import com.example.music.backend.user.service.UserService;
+import com.example.music.backend.verification.domain.VerificationToken;
+import com.example.music.backend.verification.service.VerificationTokenService;
+import org.springframework.context.MessageSource;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.WebRequest;
+
+import java.util.Calendar;
+import java.util.Locale;
+
+@RestController
+public class RegistrationController {
+
+    private final UserService userService;
+    private final MessageSource messages;
+    private final VerificationTokenService tokenService;
+
+    public RegistrationController(UserService userService, MessageSource messages, VerificationTokenService tokenService) {
+        this.userService = userService;
+        this.messages = messages;
+        this.tokenService = tokenService;
+    }
+
+    @GetMapping("/registrationConfirm.html")
+    public String confirmRegistration
+            (WebRequest request, Model model, @RequestParam("token") String token) {
+        Locale locale = request.getLocale();
+
+        VerificationToken verificationToken = tokenService.getVerificationToken(token);
+
+        if(verificationToken == null) {
+            String message = messages.getMessage("auth.message.expired", null, locale);
+            model.addAttribute("message", message);
+            return "badUser";
+        }
+
+        User user = verificationToken.getUser();
+        Calendar cal = Calendar.getInstance();
+
+        if(verificationToken.getExpiryDate().getTime() - cal.getTime().getTime() <= 0) {
```

Comment:
Question from previous PR

Reply:
the comment from the previous PR was not saved, so I don't know what needs to be fixed

Reply:
Investigate better!
[https://github.com/kajima-genius/music/pull/1#discussion_r661526199](url)
[https://github.com/kajima-genius/music/pull/1#discussion_r661538614](url)
[https://github.com/kajima-genius/music/pull/1#discussion_r661541272](url)
[https://github.com/kajima-genius/music/pull/1#discussion_r661526199](url)

Reply:
replaced with before ()